lib/Widgets/AbstractSettingsPage.vala

```diff
@@ -0,0 +1,82 @@
+/*
+* Copyright (c) 2017 elementary LLC. (https://elementary.io)
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation, either version 2.1 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Library General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+/**
+ * AbstractSettingsPage is a #Gtk.ScrolledWindow subclass with properties used
+ * by other Granite settings widgets.
+ */
+public abstract class Granite.SettingsPage : Gtk.ScrolledWindow {
+    protected string _icon_name;
+    protected string _title;
+
+    /**
+     * Used to display a status icon overlayed on the display_widget in a Granite.SettingsSidebar
+     */
+    public enum StatusType {
+        ERROR,
+        OFFLINE,
+        SUCCESS,
+        WARNING,
+        NONE
+    }
+
+    /**
+     * Selects a colored icon to be displayed in a Granite.SettingsSidebar
+     */
+    public StatusType status_type { get; set; default = StatusType.NONE; }
+
+    /**
+     * A widget to display in place of an icon in a Granite.SettingsSidebar
+     */
+    public Gtk.Widget? display_widget { get; construct; }
+
+    /**
+     * A header to be sorted under in a Granite.SettingsSidebar
+     */
+    public string? header { get; construct; }
+
+    /**
+     * A status string to be displayed underneath the title in a Granite.SettingsSidebar
+     */
+    public string status { get; set; }
+
+    /**
+     * An icon name to be displayed in a Granite.SettingsSidebar
+     */
+    public string? icon_name {
```

Comment:
I still cannot understand why there is `_icon_name` field. This `icon_name` property should be `virtual` and classes like `SimpleSettingsPage` should override it. Currently this code throws warnings like:
```
AbstractSimpleSettingsPage.vala:70.5-70.27: warning: Granite.SimpleSettingsPage.icon_name hides inherited property `Granite.SettingsPage.icon_name'. Use the `new' keyword if hiding was intentional
AbstractSimpleSettingsPage.vala:85.5-85.23: warning: Granite.SimpleSettingsPage.title hides inherited property `Granite.SettingsPage.title'. Use the `new' keyword if hiding was intentional
```

Reply:
Then perkaps add `new` in the overriding declaration?

Reply:
Again, it cannot be an override. overriding breaks the cast

Reply:
Fair enough - should this then be escalated upstream to Vala?